inst/pages/21_microbiome_community.qmd

```diff
@@ -223,11 +231,79 @@ pheatmap(mat, annotation_row = taxa_clusters,
          color = colors)
 ```
 
+### Composition heatmap using the NeatMap method {#sec-composition-neatmap}
+
+Another method to visualize community composition is by plotting a
+NeatMap, which means we use radial theta sorting when plotting the
+heatmap [@Rajaram2010]. The `getNeatOrder` function in the `miaViz`
+package allows us to achieve this. This method sorts data points based
+on their angular position in a 2D space, typically after an ordination
+technique such as PCA or NMDS has been applied.
+
+The `getNeatOrder` method calculates the angle (theta) for each point
+relative to the centroid and sorts data points based on these theta
+values in ascending order. This approach preserves the relationships
+between data points according to the ordination method's spatial
+configuration, rather than relying on hierarchical clustering.
+
+First, we will load the necessary libraries and load `GlobalPatterns`, which is a `TreeSE` object.
+
+```{r setup-libraries-data}
+# Load the required libraries
+library(mia)
+library(miaViz)
+library(scater)
+library(ComplexHeatmap)
+library(circlize)
+
+# Load the data
+data("GlobalPatterns", package="mia")
+tse <- GlobalPatterns
+```
+
+Now, we'll create the NeatMap using the `ComplexHeatmap` package and the `getNeatOrder` function.
+
+```{r NeatMap}
+# Add a pseudocount to the counts data
+assay(tse, "counts") <- assay(tse, "counts") + 1
+
+# Transform the samples into relative abundances using CLR
+tse <- transformAssay(tse, assay.type = "counts", method = "clr", MARGIN = "samples", name = "clr")
+
+# Transform the features (taxa) into zero mean, unit variance (standardize transformation)
+tse <- transformAssay(tse, assay.type = "clr", method = "standardize", MARGIN = "features", name = "standardized")
+
+# Perform PCA on the dataset
+reducedDim(tse, "PCA") <- calculatePCA(tse, ncomponents = 10, assay.type = "standardized")
+
+# Sort by radial theta using the first two principal components
+sorted_order <- getNeatOrder(reducedDim(tse, "PCA")[, c(1, 2)], centering = "mean")
+tse <- tse[, sorted_order]
+
+# Define the color function and cap the colors at [-5, 5]
+col_fun <- colorRamp2(c(-5, 0, 5), c("blue", "white", "red"))
+
+# Create the heatmap
+heatmap <- Heatmap(assay(tse, "standardized"),
+                   name = "NeatMap",
+                   col = col_fun,
+                   cluster_rows = FALSE,  # Do not cluster rows
+                   cluster_columns = FALSE,  # Do not cluster columns
+                   show_row_dend = FALSE,
+                   show_column_dend = FALSE,
+                   row_names_gp = gpar(fontsize = 4),
+                   column_names_gp = gpar(fontsize = 6),
+                   heatmap_width = unit(20, "cm"),
+                   heatmap_height = unit(15, "cm"))
+
+heatmap
```

Comment:
I would prefer sechm here

Reply:
Which heatmap do you prefer - 

This one uses the diet colData add in 
<img width="934" alt="Screenshot 2024-06-27 at 13 33 55" src="https://github.com/microbiome/OMA/assets/116303632/1af1167a-b6bc-4444-b04f-cef465892ca9">

Then this one has gaps at diet, which looks nicer, but messes up the sample ordering
<img width="589" alt="Screenshot 2024-06-27 at 13 35 59" src="https://github.com/microbiome/OMA/assets/116303632/3c4b14be-2b01-4c57-a424-7cd0327e0841">

Then this one just removes the whole diet thing
<img width="811" alt="Screenshot 2024-06-27 at 13 37 25" src="https://github.com/microbiome/OMA/assets/116303632/3bcca697-cd7c-4a3b-b444-f67b887fd605">

Reply:
(size is the same it's just how big the plot window is)

Reply:
If you split samples based on variable (diet), does it then reorder the data? "neatorder" is not preserved?

I think the last one is the best. Check that it follows the order of samples got from getNeatOrder. Also the color scale should be scaled so that 0 is in the middle.

Reply:
Value +3 seems that it is 0 while value -3 is very low in the scale

Reply:
ok sure i'll scale the colour

Reply:
<img width="608" alt="Screenshot 2024-06-28 at 14 43 51" src="https://github.com/microbiome/OMA/assets/116303632/364d1154-48a2-4614-99fd-b46a084fd4c8">

looks like this now